### PR TITLE
TASK: Improve phpstan coverage

### PR DESCRIPTION
**Pull request by mhsdesign**
Best reviewed commit by commit ;)

**Upgrade instructions**

**Review instructions**

**Checklist**

- [ ] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [ ] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

### CODE REVIEW

File: Neos.ContentRepository.Core/Classes/DimensionSpace/InterDimensionalVariationGraph.php [hunk 1]
```diff
@@ -26,49 +26,49 @@ class InterDimensionalVariationGraph
...
     /**
      * Primary generalization dimension space points, indexed by specialization hash
      *
      * @var array<string,DimensionSpacePoint>
      */
-    protected ?array $primaryGeneralizations = null;
+    protected array $primaryGeneralizations;
 
-    protected ?int $weightNormalizationBase = null;
+    protected int $weightNormalizationBase;
```
Comment (bwaidelich) [inline Neos.ContentRepository.Core/Classes/DimensionSpace/InterDimensionalVariationGraph.php]: can we make the class `final` and those `private` while we're at it?
My IDE even complaints about them being unused like that

File: Neos.ContentRepository.Core/Classes/DimensionSpace/InterDimensionalVariationGraph.php [hunk 2]
```diff
@@ -93,10 +93,9 @@ public function getDimensionSpacePoints(): DimensionSpacePointSet
...
     public function getWeightedDimensionSpacePoints(): array
     {
-        if (is_null($this->weightedDimensionSpacePoints)) {
+        if (!isset($this->weightedDimensionSpacePoints)) {
             $this->initializeWeightedDimensionSpacePoints();
         }
-        /** @var array<string,WeightedDimensionSpacePoint> $weighedDimensionSpacePoints */
         $weighedDimensionSpacePoints = $this->weightedDimensionSpacePoints;
 
         return $weighedDimensionSpacePoints;
```
Comment (bwaidelich) [inline Neos.ContentRepository.Core/Classes/DimensionSpace/InterDimensionalVariationGraph.php]: The local variable is totally useless (and contains a typo)
```suggestion
        return $this->weightedDimensionSpacePoints;
```

File: Neos.ContentRepository.Core/Classes/DimensionSpace/InterDimensionalVariationGraph.php [hunk 3]
```diff
@@ -225,6 +223,8 @@ protected function initializeVariations(): void
      * @param array<string,int> $normalizedVariationWeights
      * @param array<string,array<string,DimensionSpacePoint>>& $indexedGeneralizations
      * @param array<string,array<string,DimensionSpacePoint>>& $indexedSpecializations
+     * @param-out array<string,array<string,DimensionSpacePoint>> $indexedGeneralizations
+     * @param-out array<string,array<string,DimensionSpacePoint>> $indexedSpecializations
```
Comment (bwaidelich) [inline Neos.ContentRepository.Core/Classes/DimensionSpace/InterDimensionalVariationGraph.php:225]: I didn't know the `@param-out` annotation, nice :)

File: Neos.ContentRepository.Core/Classes/DimensionSpace/InterDimensionalVariationGraph.php [hunk 4]
```diff
@@ -237,8 +237,8 @@ protected function initializeVariationsForDimensionSpacePointPair(
         if (isset($indexedGeneralizations[$generalization->getIdentityHash()])) {
             $generalizations = $indexedGeneralizations[$generalization->getIdentityHash()];
             foreach ($generalizations as $parentGeneralizationHash => $parentGeneralization) {
-                /** @var WeightedDimensionSpacePoint $weighedParent */
                 $weighedParent = $this->getWeightedDimensionSpacePointByHash($parentGeneralizationHash);
+                assert($weighedParent !== null);
                 $generalizationsToProcess[$parentGeneralizationHash] = $weighedParent;
```
Comment (bwaidelich) [inline Neos.ContentRepository.Core/Classes/DimensionSpace/InterDimensionalVariationGraph.php]:
```suggestion
                $weightedParent = $this->getWeightedDimensionSpacePointByHash($parentGeneralizationHash);
                assert($weightedParent !== null);
                $generalizationsToProcess[$parentGeneralizationHash] = $weightedParent;
```

File: Neos.ContentRepository.Export/src/Asset/ValueObject/SerializedImageVariant.php [hunk 5]
```diff
@@ -71,11 +71,6 @@ public static function fromArray(array $array): self
 
     public function matches(ImageVariant $imageVariant): bool
     {
-        if (self::fromImageVariant($imageVariant)->toJson() !== $this->toJson()) {
-            \Neos\Flow\var_dump(self::fromImageVariant($imageVariant)->toJson());
-            \Neos\Flow\var_dump($this->toJson());
-            exit;
-        }
```
Comment (bwaidelich) [inline Neos.ContentRepository.Export/src/Asset/ValueObject/SerializedImageVariant.php:78]: whoo, hello there

**Review by bwaidelich — APPROVED** [review]
Great work, you really knocked out PHPStan!
+1 by reading.

I added some comments, but they can safely be ignored or addressed in a separate PR

**Review by kitsunet — APPROVED** [review]

**Pull request merged**